source/patterns/@aws-solutions-constructs/aws-apigateway-sagemakerendpoint/lib/index.ts

```diff
@@ -0,0 +1,163 @@
+/**
+ *  Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+// Imports
+import * as api from '@aws-cdk/aws-apigateway';
+import * as iam from '@aws-cdk/aws-iam';
+import * as defaults from '@aws-solutions-constructs/core';
+import { Aws, Construct } from '@aws-cdk/core';
+import { LogGroup } from '@aws-cdk/aws-logs';
+
+/**
+ * @summary The properties for the ApiGatewayToSageMakerEndpointProps class.
+ */
+export interface ApiGatewayToSageMakerEndpointProps {
+    /**
+     * Optional user-provided props to override the default props for the API Gateway.
+     *
+     * @default - Default properties are used.
+     */
+    readonly apiGatewayProps?: api.RestApiProps,
+
+    /**
+     * Name of the deployed SageMaker inference endpoint.
+     *
+     * @default - None.
+     */
+    readonly endpointName: string,
```

Comment:
The path for the integration has to be `endpoints/<sagemaker-endpoint-name>/invocations`, so if ARN was used we'd have to parse it. Is there an example of this in Constructs right now?

Reply:
Check `aws-apigateway-iot` pattern

Reply:
So I looked, but it's a subdomain there (e.g. `ab123cdefghij4l-ats.iot.ap-south-1.amazonaws.com`), which is different than the ARN. I could do something like this:

```javascript
// Parse endpoint name
try {
    const components = Arn.parse(props.sageMakerEndpoint);
    
    // This line is required since resourceName is string | undefined
    if (!components.resourceName) {
        throw new Error('Provided ARN does not contain resource name');
    }

    this.sageMakerEndpointName = components.resourceName;
} catch {
    this.sageMakerEndpointName = props.sageMakerEndpoint;
}
```

Would that be enough? I'm not sure if the complexity added is worth it.

Reply:
Probably not worth the complexity, let's make sure the comments clearly say it expects the endpoint Name and not the ARN of the endpoint

Reply:
Both the interface comment and README file mention name, not ARN (_Name of the deployed SageMaker inference endpoint_). Is there anywhere I should change it?